src/storage/src/hummock/event_handler/hummock_event_handler.rs

```diff
@@ -90,20 +92,25 @@ impl BufferTracker {
     }
 }
 
+type InstanceId = u64;
+pub type ReadVersionMappingType =
+    RwLock<HashMap<TableId, HashMap<InstanceId, Arc<RwLock<HummockReadVersion>>>>>;
+
 pub struct HummockEventHandler {
-    sstable_id_manager: SstableIdManagerRef,
     hummock_event_rx: mpsc::UnboundedReceiver<HummockEvent>,
     pending_sync_requests: BTreeMap<HummockEpoch, oneshot::Sender<HummockResult<SyncResult>>>,
-
-    // TODO: replace it with hashmap<id, read_version>
-    read_version: Arc<RwLock<HummockReadVersion>>,
+    read_version_mapping: Arc<ReadVersionMappingType>,
 
     version_update_notifier_tx: Arc<tokio::sync::watch::Sender<HummockEpoch>>,
     seal_epoch: Arc<AtomicU64>,
     pinned_version: Arc<ArcSwap<PinnedVersion>>,
     write_conflict_detector: Option<Arc<ConflictDetector>>,
 
     uploader: HummockUploader,
+
+    instance_id_generator: Arc<AtomicU64>,
```

Comment:
Seems that the id generator is used only in the event handler. If so, we don't have to use an atomic variable.